File: missing-number/missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        x1 = 0
        x2 = 0
        n = len(nums)
        for i in range(n):
            x1 ^= nums[i]
            x2 ^= i+1
        return x1^x2

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int x1 = 0, x2 = 0;
        for(int i = 0; i<nums.length; i++){
            x1^=nums[i];
            x2^=(i+1);
        }
        return x1^x2;
    }
}